gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,355 @@
+      this.projectNumber = info.projectNumber;
+      this.state = info.state;
+      this.createTimeMillis = info.createTimeMillis;
+      this.parent = info.parent;
+    }
+
+    /**
+     * Set the user-assigned name of the project.
+     *
+     * This field is optional and can remain unset. Allowed characters are: lowercase and uppercase
+     * letters, numbers, hyphen, single-quote, double-quote, space, and exclamation point. This
+     * field can be changed after project creation.
+     */
+    public Builder name(String name) {
+      this.name = firstNonNull(name, Data.<String>nullOf(String.class));
+      return this;
+    }
+
+    /**
+     * Set the unique, user-assigned ID of the project.
+     *
+     * The ID must be 6 to 30 lowercase letters, digits, or hyphens. It must start with a letter.
+     * Trailing hyphens are prohibited. This field cannot be changed after the server creates the
+     * project.
+     */
+    public Builder projectId(String projectId) {
+      this.projectId = projectId;
+      return this;
+    }
+
+    /**
+     * Add a label associated with this project.
+     *
+     * See {@link #labels} for label restrictions.
+     */
+    public Builder addLabel(String key, String value) {
+      this.labels.put(key, value);
+      return this;
+    }
+
+    /**
+     * Remove a label associated with this project.
+     */
+    public Builder removeLabel(String key) {
+      this.labels.remove(key);
+      return this;
+    }
+
+    /**
+     * Clear the labels associated with this project.
+     */
+    public Builder clearLabels() {
+      this.labels.clear();
+      return this;
+    }
+
+    /**
+     * Set the labels associated with this project.
+     *
+     * Label keys must be between 1 and 63 characters long and must conform to the following regular
+     * expression: [a-z]([-a-z0-9]*[a-z0-9])?. Label values must be between 0 and 63 characters long
+     * and must conform to the regular expression ([a-z]([-a-z0-9]*[a-z0-9])?)?. No more than 256
+     * labels can be associated with a given resource. This field can be changed after project
+     * creation.
+     */
+    public Builder labels(Map<String, String> labels) {
+      this.labels = Maps.newHashMap(checkNotNull(labels));
+      return this;
+    }
+
+    Builder projectNumber(Long projectNumber) {
+      this.projectNumber = projectNumber;
+      return this;
+    }
+
+    Builder state(State state) {
+      this.state = state;
+      return this;
+    }
+
+    Builder createTimeMillis(Long createTimeMillis) {
+      this.createTimeMillis = createTimeMillis;
+      return this;
+    }
+
+    Builder parent(ResourceId parent) {
+      this.parent = parent;
+      return this;
+    }
+
+    public ProjectInfo build() {
+      return new ProjectInfo(this);
+    }
+  }
+
+  ProjectInfo(Builder builder) {
+    this.name = builder.name;
+    this.projectId = checkNotNull(builder.projectId);
```

Comment:
Not needed if checked in the `Builder`. right now it is not consistent. some checked in Builder and some in constructor.